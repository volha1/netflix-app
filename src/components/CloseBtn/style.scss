@import '../../common/scss/base.scss';

.close-btn::after {
  display: flex;
  position: relative;
  justify-content: right;
  content: '\00d7';
  font-size: 60px;
  font-weight: 300;
  cursor: pointer;
  top: -5px;
  right: 5px;
  line-height: 0;
  z-index: 10;
}
