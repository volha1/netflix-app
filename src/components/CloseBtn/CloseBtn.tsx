import React, { ReactElement } from 'react';
import './style.scss';

type CloseBtnProps = {
  handleClose: () => void;
};

const CloseBtn = ({ handleClose }: CloseBtnProps): ReactElement => <div className="close-btn" onClick={handleClose} />;

export default CloseBtn;
