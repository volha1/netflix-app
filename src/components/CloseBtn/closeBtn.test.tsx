import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import CloseBtn from './CloseBtn';

describe('CloseBtn test', () => {
  const onClose = jest.fn();

  it('CloseBtn snapshot', () => {
    const component = renderer.create(<CloseBtn onClose={onClose} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('onClose works', async () => {
    render(<CloseBtn onClose={onClose} />);

    await userEvent.click(screen.getByTestId('close-btn'));

    expect(onClose).toHaveBeenCalled();
  });
});
