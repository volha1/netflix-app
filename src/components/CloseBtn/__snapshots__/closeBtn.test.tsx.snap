// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseBtn test CloseBtn snapshot 1`] = `
<div
  className="close-btn"
  data-testid="close-btn"
  onClick={[MockFunction]}
/>
`;
